tsfile/src/main/java/org/apache/iotdb/tsfile/encoding/encoder/RegularDataEncoder.java

```diff
@@ -0,0 +1,438 @@
+      if (isMissingPoint) {
+        writeBitmap(out);
+      }
+      // write header
+      writeHeaderToBytes();
+
+      reset();
+      writeIndex = -1;
+    }
+
+    @Override
+    protected void reset() {
+      blockSize = BLOCK_DEFAULT_SIZE;
+      minDeltaBase = Integer.MAX_VALUE;
+      isMissingPoint = false;
+      isLastPack = false;
+      firstValue = 0;
+      previousValue = 0;
+    }
+
+    @Override
+    protected void writeHeader() throws IOException {
+      out.write(BytesUtils.intToBytes(minDeltaBase));
+      out.write(BytesUtils.intToBytes(firstValue));
+    }
+
+    @Override
+    public void encode(int value, ByteArrayOutputStream out) {
+      try {
+        encodeValue(value, out);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    @Override
+    public int getOneItemMaxSize() {
+      return 4;
+    }
+
+    @Override
+    public long getMaxByteSize() {
+      // The meaning of 20 is: identifier(4)+bitmapLength(4)+index(4)+minDeltaBase(4)+firstValue(4)
+      return (long) 20 + (writeIndex * 2 / 8) + (writeIndex * 4);
+    }
+
+    /**
+     * input a integer or long value.
+     *
+     * @param value value to encode
+     * @param out   - the ByteArrayOutputStream which data encode into
+     */
+    public void encodeValue(int value, ByteArrayOutputStream out) throws IOException {
+      if (writeIndex == -1) {
+        data = new int[blockSize];
+        writeIndex = 0;
+      }
+      data[writeIndex++] = value;
+      if (writeIndex == blockSize) {
+        checkMissingPoint(out);
+      }
+    }
+
+    private void checkMissingPoint(ByteArrayOutputStream out) throws IOException {
+      // get the new regular data if the missing point exists in the original data
+      data = getRegularData(data);
+      firstValue = data[0];
+      if (isMissingPoint) {
+        writeIndex = data.length;
+      }
+      if (!isLastPack) {
+        flush(out);
+      }
+    }
+
+    private void writeBitmap(ByteArrayOutputStream out) throws IOException {
+      // generate bitmap
+      data2Diff(missingPointData);
+      byte[] bsArr = bitmap.toByteArray();
+      out.write(BytesUtils.intToBytes(bsArr.length));
+      out.write(bsArr);
+    }
+
+    private int[] getRegularData(int[] data) {
+      if (writeIndex > 1) {
+        previousValue = data[0];
+        minDeltaBase = data[1] - data[0];
+        // calculate minimum elapsed of the data and check whether the missing point exists
+        for (int i = 1; i < writeIndex; i++) {
+          calcDelta(data[i]);
+          previousValue = data[i];
+        }
+      }
+      // generate the continious data when the missing point exists
+      if (isMissingPoint) {
+        generateRegularData(data);
+        return regularData;
+      }
+      return data;
+    }
+
+    private void generateRegularData(int[] data) {
+      missingPointData = data;
+      blockSize = (int) (((data[writeIndex - 1] - data[0]) / minDeltaBase) + 1);
```

Comment:
`int / int ` returns integer value, thus `(int)` is redundant.

Reply:
Done. Thx.